File: server/routes/alerts.py
from flask import Blueprint, render_template, jsonify, request
from flask_login import login_required
from models import Alert, Employee
from extensions import db, socketio

alerts_bp = Blueprint('alerts', __name__)

@alerts_bp.route('/alerts')
@login_required
def alerts_page():
    alerts = Alert.query.order_by(Alert.created_at.desc()).limit(100).all()
    return render_template('alerts.html', alerts=alerts)

@alerts_bp.route('/api/alerts', methods=['GET'])
@login_required
def get_alerts():
    alerts = Alert.query.order_by(Alert.created_at.desc()).limit(50).all()
    return jsonify([{
        'id': a.id,
        'employee_id': a.employee_id,
        'employee_name': a.employee.name,
        'type': a.alert_type,
        'message': a.message,
        'severity': a.severity,
        'is_read': a.is_read,
        'created_at': a.created_at.isoformat()
    } for a in alerts])

@alerts_bp.route('/api/alerts/<int:alert_id>/read', methods=['POST'])
@login_required
def mark_read(alert_id):
    alert = Alert.query.get_or_404(alert_id)
    alert.is_read = True
    db.session.commit()
    return jsonify({'status': 'ok'})

@alerts_bp.route('/api/alerts/read-all', methods=['POST'])
@login_required
def mark_all_read():
    Alert.query.filter_by(is_read=False).update({'is_read': True})
    db.session.commit()
    return jsonify({'status': 'ok'})

File: node_server/routes/alerts.js
const express = require('express');
const router = express.Router();
const { queryAll, runSql } = require('../db');
const { requireLogin } = require('../middleware/auth');

router.get('/alerts', requireLogin, (req, res) => {
  const alerts = queryAll(`
    SELECT a.*, e.name as employee_name FROM alerts a
    JOIN employees e ON a.employee_id = e.id
    ORDER BY a.created_at DESC LIMIT 100
  `);
  res.render('alerts.html', { alerts });
});

router.get('/api/alerts', requireLogin, (req, res) => {
  const alerts = queryAll(`
    SELECT a.*, e.name as employee_name FROM alerts a
    JOIN employees e ON a.employee_id = e.id
    ORDER BY a.created_at DESC LIMIT 50
  `);
  res.json(alerts);
});

router.post('/api/alerts/:id/read', requireLogin, (req, res) => {
  runSql('UPDATE alerts SET is_read = 1 WHERE id = ?', [parseInt(req.params.id)]);
  res.json({ status: 'ok' });
});

router.post('/api/alerts/read-all', requireLogin, (req, res) => {
  runSql('UPDATE alerts SET is_read = 1 WHERE is_read = 0');
  res.json({ status: 'ok' });
});

module.exports = router;
